contracts/src/v0.7/tests/UpkeepPerformCounterRestrictive.sol

```diff
@@ -7,18 +7,29 @@ contract UpkeepPerformCounterRestrictive {
   uint256 public nextEligible = 0;
   uint256 public testRange;
   uint256 public averageEligibilityCadence;
-  uint256 count = 0;
+  uint256 public checkGasToBurn;
+  uint256 public performGasToBurn;
+  mapping(bytes32 => bool) public dummyMap; // used to force storage lookup
+
+  uint256 private count = 0;
 
   constructor(uint256 _testRange, uint256 _averageEligibilityCadence) {
     testRange = _testRange;
     averageEligibilityCadence = _averageEligibilityCadence;
   }
 
   function checkUpkeep(bytes calldata data) external view returns (bool, bytes memory) {
-    return (eligible(), bytes(""));
+    uint256 startGas = gasleft();
+    bool dummy;
+    // burn gas
+    while (startGas - gasleft() < checkGasToBurn) {
+      dummy = dummy && dummyMap[blockhash(block.number)]; // arbitrary state reads
```

Comment:
maybe do the same operation as performUpkeep (dummyMap[blockhash(block.number)] = false;) to keep things simple?